client/gutenberg/editor/controller.js

```diff
@@ -192,14 +166,9 @@ export const redirect = async ( context, next ) => {
 	} = context;
 	const tmpState = getState();
 	const selectedEditor = getSelectedEditor( tmpState, getSelectedSiteId( tmpState ) );
-	const checkPromises = [];
 	if ( ! selectedEditor ) {
-		checkPromises.push( waitForSiteIdAndSelectedEditor( context ) );
-	}
-	if ( ! isPreferredEditorViewAvailable( tmpState ) ) {
-		checkPromises.push( waitForPreferredEditorView( context ) );
+		await waitForSiteIdAndSelectedEditor( context );
```

Comment:
I think we no longer need to wait for the selected editor either. This was needed in the past since, depending on the value we could redirect to either Gutenberg or the (now extinct) Calypso editor (see https://github.com/Automattic/wp-calypso/commit/ddf3ad0df5c3a33e69c5eb60e5c5a1b1710efbd5#diff-0a9a19e0cb60f14e80f3da28550ac16061317cdaf9fe28735e585ab703f8da23).

We now always redirect to Gutenberg regardless of the selected editor (see [`getEditorUrl`](https://github.com/Automattic/wp-calypso/blob/5c24e9e99148a6e08b1f789752dc4406c1033cb3/client/state/selectors/get-editor-url.ts#L7-L25))